components/script/dom/bindings/structuredclone.rs

```diff
@@ -67,16 +67,41 @@ unsafe fn read_length(r: *mut JSStructuredCloneReader)
   return length as usize;
 }
 
+struct StructuredCloneWriter {
+    w: *mut JSStructuredCloneWriter,
+}
+
+impl StructuredCloneWriter {
+    unsafe fn write_slice(&self, v: &[u8]) {
+        let type_length = v.len();
+        write_length(self.w, type_length);
+        assert!(JS_WriteBytes(self.w, v.as_ptr() as *const raw::c_void, type_length));
+    }
```

Comment:
Let's add a corresponding `write_str` which accepts a `&str` argument and calls `write_slice(s.as_byte())`.